### DISCUSSION TITLE
Making a combined CLs plots for various signal strength hypotheses

### DISCUSSION THREAD

**Discussion by ssaha1234** [scikit-hep/pyhf#1838] [Q&A]
Dear pyhf members, 
      I am creating a combination plot using CLs values across various hypotheses mass points for my analysis, I have made one CLs plot per mass point, something like this - 
<img width="692" alt="Screen Shot 2022-04-04 at 9 10 32 AM" src="https://user-images.githubusercontent.com/86737921/161550907-34d7697a-ef4f-4135-b7a7-15b9534e9446.png">

I would now like to make a plot, by combining all the masses I have (~12 such CLs plots above for 12 mass points), as in the plot here -

<img width="532" alt="Screen Shot 2022-04-04 at 9 08 39 AM" src="https://user-images.githubusercontent.com/86737921/161550561-7dba033d-621f-4e8f-b76a-248473a28623.png">
 
Is there a way to do this easily with cabinetry, for example ? I am looking at the example here, for reference - https://github.com/cabinetry/cabinetry-tutorials/blob/08a0363db0585d12bdcbc9600c9e139351142e8b/example.ipynb 

Thanks ! 
Best, 
Shreya

**Comment by alexander-held** [comment]
Hi Shreya, neither `pyhf` nor `cabinetry` currently have an interface for this kind of plot. It would be great to implement this somewhere centrally though (perhaps even `mplhep` could fit). I am sure that some people have already made that kind of plot with `matplotlib`, so perhaps we can build on existing implementations.

**Reply by alexander-held** [reply]
Related plots that might also be useful to implement: https://github.com/iris-hep/as-user-facing/blob/master/fit-visualization/exclusion-limits.md

**Reply by matthewfeickert** [reply]
@ssaha1234 We probably aren't going to prioritize this any time soon either on `pyhf`'s side of things from the core dev team side as the visualization module is currently under `contrib` (`pyhf.contrib.viz`) and we would prefer to ideally have all `visualization` factored out into another library (eventually). That being said, the `contrib` name is supposed to be suggestive of the fact that we would be open to contributions from community members that want to implement this. :+1: 

> Related plots that might also be useful to implement: https://github.com/iris-hep/as-user-facing/blob/master/fit-visualization/exclusion-limits.md

To emphasize Alex's point here, we **do** think these sorts of plots are quite important (we're users of them ourselves)! The primary point is that `pyhf` and `cabinetry` are libraries and _not_ frameworks, so they should avoid scope creep that keeps them from achieving and improving on their primary goals, especially when other libraries in the Scikit-HEP ecosystem already specialize in visualization.

**Reply by ssaha1234** [reply]
Hi @alexander-held and @matthewfeickert, 
  Thanks a lot for pointing out to the scripts ! I can share the script I would be writing, once I have it working :)
Best, 
Shreya

**Comment by kratsg** [comment]
I also have this notebook (https://gist.github.com/kratsg/4ff8cb2ded3b25552ff2f51cd6b854dc) which should help on some level.

**Comment by souravampire** [comment]
Hi @ssaha1234, did you solve it? I am stuck in exactly the same problem. I would appreciate it if you could share your progress.
